lib/log_parser.rb

```diff
@@ -28,24 +30,53 @@ def openfile
     file.close unless file.nil?
   end
 
-    def finding_players
+  def finding_players
       players = []
 
-      for i in 0..@file_data.length-1 do
-        if @file_data[i].include? "killed"
-          current_line = @file_data[i]
-          player_1 = current_line.split("killed")[0].split(":").last.strip
-          player_2 = current_line.split("killed")[1].split("by").first.strip
-          if !players.include? player_1
-            players.push(player_1)
-          end
-          if !players.include? player_2
+    for i in 0..@file_data.length-1 do
+      if @file_data[i].include? "killed"
+        current_line = @file_data[i]
+        player_1 = current_line.split("killed")[0].split(":").last.strip
+        player_2 = current_line.split("killed")[1].split("by").first.strip
+        if !players.include? player_1
+          players.push(player_1)
+        end
+        if !players.include? player_2
             players.push(player_2)
+        end
+      end
+    end
+    players.delete("<world>")
+    return players
+  end
+
+  def kills_count
+    kills_hash = {}
+
+    players_name = @players
+
+  (0..players_name.length - 1).each do |i|
+    kills_hash[players_name[i].to_s] = 0
+    end
+    @file_data.each do |line|
+        if line.include? "killed"
+          current_line = line
+          player_1 = current_line.split("killed")[0].split(":").last.strip
```

Comment:
Pode usar diretamente o `line`, nem precisa dessa variável auxiliar não:

```suggestion
          player_1 = line.split("killed")[0].split(":").last.strip
```